modules/core/src/lib/attribute.js

```diff
@@ -206,18 +208,26 @@ export default class LayerAttribute extends Attribute {
           this.update({constant: false, buffer});
           state.needsRedraw = true;
         }
-      } else {
-        const ArrayType = glArrayFromType(this.type || GL.FLOAT);
-        if (!(buffer instanceof ArrayType)) {
-          throw new Error(`Attribute ${this.id} must be of type ${ArrayType.name}`);
-        }
+      } else if (this.value !== buffer) {
```

Comment:
Looks like `setExternalBuffer` now takes arrays and not just buffers? ...

Maybe break this part into a separate function and consider naming to help tame complexity/guide maintainers? 

I'm surprised the linter doesn't complain about max-statements, maybe we have some linter issue?

Reply:
I'm fine with adding a check for typed array here.

This change is to avoid throwing an error if `Uint8Array`/`Float32Array` is provided instead of `Uint8ClampedArray` for colors, or `Uint16Array` is provided instead of `Uint32Array` for indices. The old treatment did not work well as we try to decouple loading from the layer itself.